src/lib/statistics/chunk_statistics/counting_quotient_filter.hpp

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <vector>
+#include <string>
+
+#include "cqf2.hpp"
+#include "cqf4.hpp"
+#include "cqf8.hpp"
+#include "cqf16.hpp"
+#include "cqf32.hpp"
+#include "types.hpp"
+
+#include "storage/base_segment.hpp"
+#include "abstract_filter.hpp"
+
+
+
+
+namespace opossum {
+
+/* Counting Quotient Filters allow you to keep track of which values are present in a segment and how often. Filters
+work approximately. If a membership query yields a positive result, the value is probably present but there is
+a chance of a false positive. If the query delivers a negative result, the item is guaranteed to not be contained.
+In the same way, items can be over counted but not under counted.
+CQF can be configured with quotient size, which determines the number of slots, and the remainder size, which
+corresponds to the slot size. At this time, the remainder size must be 2, 4, 8, 16 or 32. */

```

Comment:
Could you explain quotient_size and remainder_size a little bit more? I would have no idea what reasonable values are.

Reply:
I added another comment that explains the sizing.